FormidableLabs/spectacle
# Rewrite/mdx notes parsing

## Description
I've updated the mdx webpack loader to extract all text from the mdx file that is prepended with `Notes: `. After the text is extracted, it is run through the mdx compiler in the same way that slide content is processed. The output is an array of nullable mdx components. The item in the array will be null instead of a component if the original slide markdown did not contain any notes. The notes components array will be the same length as the slide components array.

You can import the mdx notes components one of two ways:
1. In the case of cloning the spectacle project and running it via `yarn start`, you'll use `import {notes} from './path/to/my/file.mdx'`.
2. In the case of running the CLI on an mdx file, there is a webpack alias to the user input file called `spectacle-user-mdx`. There is an example import in `mdx-slides/index.js`.

## Type of Change
New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

To test new functionality:

1. check out this branch locally
2. run npm link in the spectacle directory
3. move to any directory you like, e.g. ~/Documents/test/nest
4. create a new MDX file with notes (prepend notes with `Notes: `)
; remember that you must create any components that are imported into the MDX
5. run spectacle -s <path to your mdx file> to boot up the Spectacle presentation
